client/src/components/kvEditor/KVEditorRow.tsx

```diff
@@ -155,13 +155,30 @@ function KVEditorRow({
     }
   }
 
+  const eventHandlers = EditorView.domEventHandlers({
+    paste(event) {
+      event.preventDefault();
+      if (!event.clipboardData) {
+        return;
+      }
+      const text = event.clipboardData.getData('text');
+      const sanitized = text.replace(/(\r\n|\n|\r)/gm, '');
+      const selection = window.getSelection();
+      if (!selection || !selection.rangeCount) return;
+      selection.deleteFromDocument();
+      selection.getRangeAt(0).insertNode(document.createTextNode(sanitized));
```

Comment:
Is this really the only way to intercept pasting in codemirror? This looks very complicated TBH. I would have thought there is sth like `beforeChange` or sth like that where you have a function that receives a string and returns a string. Did you check for sth like this?

Reply:
I get what you mean. 
The problem is, we only want to alter the clipboard when pasting.
That means we have to hook into the paste event.
With codemirror this is done like implemented here.

Within the hook, we have access to the objects `ClipboardEvent` and the `EditorView` object.
Atm, I don't see another way than getting the information from the event and then manually updating the target.

As an alternative we could sanitize every input with some `beforeChange` hook that surely exists. But, running on every input is probably not what we want.

Reply:
I am just worried about all the stuff regarding selections and inserting nodes. Do you have some reference to why you are doing it like this? I had hoped that there is some way to just intercept the event, sanitize the value, and the continue the paste process as usual.

Reply:
It's not ideal, but I didn't find a hook to use.

In the newest commit, I succeeded to intercept the event, manipulate it and dispatch it again.
Looks much better than the previous, manual way.